plugins/parquet/parquetembed.cpp

```diff
@@ -497,169 +328,282 @@ arrow::Status ParquetHelper::processReadFile()
 {
-    return partition;
+    if (scanner)
+        return !(totalRowsProcessed >= totalRowCount);
+    else
+        return !(tablesProcessed >= tableCount && rowsProcessed >= rowsCount);
 }
 
 /**
- * @brief Returns the maximum size of the row group set by the user. Default is 1000.
+ * @brief Iterates to the correct starting RecordBatch in a partitioned dataset.
  *
- * @return int Maximum size of the row group.
+ * @return arrow::Result A pointer to the current table.
  */
-__int64 ParquetHelper::getMaxRowSize()
+arrow::Result<std::shared_ptr<arrow::Table>> ParquetReader::queryRows()
 {
-    return rowSize;
+    // If no tables have been processed find the starting RecordBatch
+    if (tablesProcessed == 0)
+    {
+        // Start by getting the number of rows in the first group and checking if it includes this workers startRow
+        __int64 offset = (*rbatchItr)->get()->num_rows();
+        while (offset < startRow)
+        {
+            rbatchItr++;
+            offset += (*rbatchItr)->get()->num_rows();
+        }
+        // If startRow is in the middle of a table skip processing the beginning of the batch
+        rowsProcessed = (*rbatchItr)->get()->num_rows() - (offset - startRow);
+    }
+    // Convert the current batch to a table
+    PARQUET_ASSIGN_OR_THROW(auto batch, *rbatchItr);
+    rbatchItr++;
+    std::vector<std::shared_ptr<arrow::RecordBatch>> toTable = {batch};
+    return std::move(arrow::Table::FromRecordBatches(std::move(toTable)));
 }
 
-char ParquetHelper::queryPartOptions()
+/**
+ * @brief Updates the current table if all the rows have been proccessed. Sets nextTable to the current TableColumns object.
+ *
+ * @param nextTable The memory address of the TableColumns object containing the current table.
+ * @return __int64 The number of rows that have been processed and the current index in the columns.
+ */
+__int64 ParquetReader::next(TableColumns *&nextTable)
 {
-    if (partOption[0] == 'W' || partOption[0] == 'w')
+    if (rowsProcessed == rowsCount)
     {
-        return 'w';
+        std::shared_ptr<arrow::Table> table;
+        if (endsWithIgnoreCase(partOption.c_str(), "partition"))
+        {
+            PARQUET_ASSIGN_OR_THROW(table, queryRows());
+        }
+        else
+        {
+            reportIfFailure(queryCurrentTable(tablesProcessed + startRowGroup)->ReadTable(&table));
+        }
+        rowsProcessed = 0;
+        tablesProcessed++;
+        rowsCount = table->num_rows();
+        splitTable(table);
     }
-    else if (partOption[0] == 'R' || partOption[0] == 'r')
+    nextTable = &parquetTable;
+    totalRowsProcessed++;
+    return rowsProcessed++;
+}
+
+/**
+ * @brief Contructs a ParquetWriter for the target location and checks for existing data.
+ *
+ * @param option The read or write option as well as information about partitioning.
+ * @param _destination The destination to write a parquet file.
+ * @param _rowSize The max row group size when creating RecordBatches for output.
```

Comment:
What is the scale?  Bytes?

Reply:
Reworded to clarify that rowSize is the maximum number of rows in a RecordBatch.

Reply:
May I suggest maxRowCountInBatch or something along those lines?

Reply:
Changed to maxRowCountInBatch. That is much more readable and causes less confusion.